ang/test/civicase/activity/factories/get-activity-feed-url.factory.spec.js

```diff
@@ -0,0 +1,151 @@
+
+      beforeEach(function () {
+        activityFeedUrlObject = getActivityFeedUrlAsObject({
+          category: category
+        });
+      });
+
+      it('filter activities by category', function () {
+        expect(activityFeedUrlObject.params.af['activity_type_id.grouping']).toBe(category);
+      });
+    });
+
+    describe('when passing an activity status type', function () {
+      beforeEach(function () {
+        activityFeedUrlObject = getActivityFeedUrlAsObject({
+          statusType: 'completed'
+        });
+      });
+
+      it('filter activities by status type', function () {
+        expect(activityFeedUrlObject.params.af.status_id).toEqual(activityStatusTypes.completed);
+      });
+    });
+
+    describe('when there are activity filters in the current route', function () {
+      beforeEach(function () {
+        $route.current.params.af = { status_id: [_.uniqueId()] };
+        activityFeedUrlObject = getActivityFeedUrlAsObject();
+      });
+
+      it('preserves the current activity filters', function () {
+        expect(activityFeedUrlObject.params.af).toEqual($route.current.params.af);
+      });
+    });
+
+    describe('when requesting filters that already exist in the current route', function () {
+      beforeEach(function () {
+        $route.current.params.af = { status_id: [_.uniqueId()] };
+        activityFeedUrlObject = getActivityFeedUrlAsObject({
+          statusType: 'completed'
+        });
+      });
+
+      it('overrides the current filters with the new ones', function () {
+        expect(activityFeedUrlObject.params.af).toEqual({
+          status_id: activityStatusTypes.completed
+        });
+      });
+    });
+
+    describe('when requesting the activities for a particular case', function () {
+      var caseId = _.uniqueId();
+
+      beforeEach(function () {
+        activityFeedUrlObject = getActivityFeedUrlAsObject({
+          caseId: caseId
+        });
+      });
+
+      it('filters activities by case id', function () {
+        expect(activityFeedUrlObject.params.caseId).toBe(caseId);
+      });
+    });
+
+    /**
+     * Returns the value returned by calling `getActivityFeedUrl` and gives
+     * some extra information: the base URL, the query parameters, and the full
+     * url path.
+     *
+     * @param {Object} urlParams the urlParams to pass to the `getActivityFeedUrl` service.
+     * @return {Object} the base url, the query params, and the full url for the activity feed.
+     */
+    function getActivityFeedUrlAsObject (urlParams) {
+      var wrappedActivityFeedUrl = getActivityFeedUrl(urlParams);
+      var activityFeedUrl = $sce.valueOf(wrappedActivityFeedUrl);
+      var activityFeedParams = extractQueryStringParams(activityFeedUrl);
+      var baseUrl = activityFeedUrl.match(/^(.+)\?/)[1];
+
+      return {
+        baseUrl: baseUrl,
+        params: activityFeedParams,
+        url: activityFeedUrl
+      };
+    }
+  });
+})(
+  CRM._,
```

Comment:
Any reason of having these namespaces in different lines?

Reply:
because on a single line they have more than 80 characters. This is not a hard rule, but it's easier to read if spread into different lines.